### ISSUE TITLE
API tidy up

### ISSUE THREAD

**Issue by adamreeve** [adamreeve/npTDMS#161]
The npTDMS API is a bit of a mess with inconsistencies and multiple ways to do things, eg. use of `channel_data` vs accessing data via the channel object. I'd like to tidy it up and introduce some breaking changes before releasing a 1.0.0 version. The plan is:

* Remove the `object`, `group_channels` and `channel_data` methods and `objects` property of `TdmsFile`
* Introduce separate `TdmsGroup` and `TdmsChannel` objects to replace the `TdmsObject` class, removing the `group` and `channel` properties and replacing them with a `name` property, and removing the `property` method
* Change the `groups` method to return a list of `TdmsGroup`
* Allow indexing into a `TdmsFile` by group name to get a group object.
* Add equivalent `channels` method to `TdmsGroup` and allow indexing by channel name to get a channel object.

With these changes, accessing a single channel's data will look like `tdms_file["group name"]["channel name"].data`, and iterating over all channels would be:
```python
for group in tdms_file.groups():
    for channel in group.channels():
        print(channel.data)
```

**Comment by adamreeve** [comment]
The new API changes were implemented in #166 but the old methods have been kept for now and will raise a deprecation warning. I'll keep this issue open until removing them completely.

In addition, slicing a `TdmsChannel` object is now the preferred way to access the data as a numpy  array, as this will work when lazily loading data too:
```python
channel_data = tdms_file["group name"]["channel name"][:]

for group in tdms_file.groups():
    for channel in group.channels():
        channel_data = channel[:]
```

**Comment by adamreeve** [comment]
Deprecated methods have been removed in #217 and I've released version 1.0.0 so am closing this
